torch/utils/data/dataloader.py

```diff
@@ -94,6 +99,11 @@ class DataLoader(object):
         worker_init_fn (callable, optional): If not ``None``, this will be called on each
             worker subprocess with the worker id (an int in ``[0, num_workers - 1]``) as
             input, after seeding and before data loading. (default: ``None``)
+        auto_collation (bool, optional): usually auto collation mode is dynamically determined,
+            (i.e. if a ``batch_sampler`` or ``batch_size`` are specified, ``auto_collation`` will 
+            be ``True`` and ``False`` otherwise). However, depending on the dataset implementation,
+            it can also be manually specified. If ``True``, auto collation mode is enabled
+            and multiple batches are merged in a single list. (default: ``None``)
```

Comment:
Can you elaborate on what this does? I'm not sure if we really need this argument. We now support batch datasets on the Python side.

Reply:
With current `DataLoader ctor` default values, `IterableDataset` dataset will always be on auto-collation mode. Because it is possible that some dataset implementations, especially the new `ChunkDataset` which extends `IterableDataset` not necessarily need auto-collation, I have added this setter (before there was a getter only) to override this behavior. Another approach would be to explicitly set `batch_size=None`.

I preferred the former approach because `ChunkDataset` ctor requires positive `batch_size` and setting `batch_size=None` to DataLoader class could be confusing for the user. In the same script we would have batch_size with a positive value and None

Reply:
Removed auto_collation in the latest revision